cmd/lotus-storage-miner/info.go

```diff
@@ -248,3 +254,71 @@ func sectorsInfo(ctx context.Context, napi api.StorageMiner) error {
 
 	return nil
 }
+
+var infoSetAddrsCmd = &cli.Command{
+	Name:  "set-addrs",
+	Usage: "set addresses that your miner can be publically dialed on",
+	Flags: []cli.Flag{
+		&cli.Int64Flag{
+			Name:  "gas-limit",
+			Usage: "set gas limit",
+			Value: 100000,
+		},
+	},
+	Action: func(cctx *cli.Context) error {
+		nodeApi, closer, err := lcli.GetStorageMinerAPI(cctx)
+		if err != nil {
+			return err
+		}
+		defer closer()
+
+		api, acloser, err := lcli.GetFullNodeAPI(cctx)
+		if err != nil {
+			return err
+		}
+		defer acloser()
+
+		ctx := lcli.ReqContext(cctx)
+
+		var addrs []abi.Multiaddrs
+		for _, a := range cctx.Args().Slice() {
+			maddr, err := ma.NewMultiaddr(a)
+			if err != nil {
+				return fmt.Errorf("failed to parse %q as a multiaddr: %w", a, err)
+			}
+
+			addrs = append(addrs, maddr.Bytes())
+		}
+
+		maddr, err := nodeApi.ActorAddress(ctx)
+		if err != nil {
+			return err
+		}
+
+		minfo, err := api.StateMinerInfo(ctx, maddr, types.EmptyTSK)
+		if err != nil {
+			return err
+		}
+
+		params, err := actors.SerializeParams(&miner.ChangeMultiaddrsParams{addrs})
+		if err != nil {
+			return err
+		}
+
+		gasLimit := cctx.Int64("gas-limit")
+
+		smsg, err := api.MpoolPushMessage(ctx, &types.Message{
+			To:       maddr,
+			From:     minfo.Worker,
+			Value:    types.NewInt(0),
+			GasPrice: types.NewInt(1),
+			GasLimit: gasLimit,
+			Method:   18,
```

Comment:
We should use enum from specs-actors here.

Reply:
i would if it were set